### PR TITLE
Adding wall to the env and to the observations

### PR DESCRIPTION
**Pull request by shamilmamedov**
1. Added a `dataclass` that describes the wall obstacle as a hyperplane
2. Added wall to gym the environment  and to the observations as `observation = [x, p_goal, w_wall, b_wall]`
3. In the `CallableMPCExpert` added a method `_parse_observation` (instead of `_separate_observation_into_state_and_goal`) that parses the obseervation into state, goal and the wall
4. Added `add_wall_obstacle` option to the `create_unified_flexiblearmenv_and_controller`
5. Changed  the angle ranges in `create_unified_flexiblearmenv_and_controller`

### CODE REVIEW

File: utils/gym_utils.py [hunk 1]
```diff
@@ -92,13 +95,23 @@ def create_unified_flexiblearmenv_and_controller(create_controller=False):
...
+        b = np.array([0., -0.15, 0.5])
+        wall = WallObstacle(w, b)
+    else:
+        wall = None
+
+    # Create environment
+    env = FlexibleArmEnv(env_options, obstacle=wall)
     # -------------------------------------
 
     if create_controller:
```
Comment (Erfi) [inline utils/gym_utils.py:117]: Here we will wait for @RudolfReiter to add the wall obstacle to the NMPC, right?

**Comment by Erfi** [comment]
LGTM! have you tested it (does data_collection script run with there changes?) if so let's merge

**Comment by shamilmamedov** [comment]
> 

It was stupid of me not to test it. Something in broken, gonna fix it now

**Comment by shamilmamedov** [comment]
There was a problem with `_parse_observation` of the `CallableMPCExpert`, fixed it with https://github.com/shamilmamedov/flexible_arm/pull/34/commits/3d0a9acd5bfd707f1b9a1413a879d94e7f2353c9

**Pull request merged**